app/controllers/reviews_controller.rb

```diff
@@ -37,9 +37,15 @@ def new
 
   def create
     @review = Review.new(review_params)
-    if @review.save
-      if params[:from]
-        redirect_to params[:from]
+    if params[:article_revision_id]
+      article_revision = ArticleRevision.find(params[:article_revision_id])
+      @review.article_revision = article_revision
+      if @review.errors[:person]
+        flash[:error] = "#{@review.errors.full_message("#{@review.person.surname}", 'ma taką samą afiliację jak jeden z autorów!')}"
+        redirect_to article_revision.submission
+        return
+      elsif @review.save
+        redirect_to article_revision.submission
```

Comment:
Ten kod raczej jest nadmiarowy. Tzn. usunięto istotny parametr params[:from], który wskazywał skąd przychodziło wywołanie (czy z widoku osoby, czy zgłoszenia). Nadto nie ma potrzeby sprawdzać jaki dokładnie błąd się pojawił, gdyż odpowiedzialny jest za to mechanizm po stronie widoku. Proszę przywrócić poprzednią wersję, która została przygotowana właśnie pod scenariusz z konfliktem w afiliacji.